server/src/test/java/org/apache/druid/query/lookup/LookupReferencesManagerTest.java

```diff
@@ -849,6 +849,26 @@ public void testCoordinatorLoadSubsetOfLookups() throws Exception
     Assert.assertFalse(lookupReferencesManager.get("testLookup3").isPresent());
   }
 
+  @Test
+  public void testAddWithLoadingSpec() throws Exception
+  {
+    LookupLoadingSpec loadingSpec = LookupLoadingSpec.loadOnly(ImmutableSet.of("testLookup1"));
+    getLookupMapForSelectiveLoadingOfLookups(loadingSpec);
+
+    LookupExtractorFactoryContainer container2 = new LookupExtractorFactoryContainer(
```

Comment:
Since this bug was observed in the Kafka task scenario, I wonder if it's possible to add a test directly in the Kafka indexing module? Not sure if there are existing UTs/ITs that use lookups, but if it's too much trouble, I think this unit test should be good.

Reply:
I can take a look and raise a follow up PR if that works.

Reply:
Yes, works for me